pkg/service/service.go

```diff
@@ -529,6 +535,20 @@ func (s *Service) createSVCInfoIfNotExist(p *lb.SVC) (*svcInfo, bool, bool,
 		s.svcByID[p.Frontend.ID] = svc
 		s.svcByHash[hash] = svc
 	} else {
+		// Reviewers: Local Redirect Policies with service matcher would have same frontend
+		// as the service clusterIP type. In such cases, if a Local redirect service
+		// exists, we shouldn't override it with clusterIP type (e.g., k8s event/sync, etc).
+		if svc.svcType == lb.SVCTypeLocalRedirect && p.Type == lb.SVCTypeClusterIP {
```

Comment:
Could it be that not only ClusterIP svc backend might match LRP svc?

Reply:
Currently, LRP service matcher only applies to Kubernetes services of type ClusterIP. It's documented in the LRP CRD spec.